tools/perf/tests/shell/test_brstack.sh

```diff
@@ -47,17 +47,17 @@ test_user_branches() {
 # first argument <arg0> is the argument passed to "--branch-stack <arg0>,save_type,u"
 # second argument are the expected branch types for the given filter
 test_filter() {
-	local filter=$1
-	local expect=$2
+	test_filter_filter=$1
+	test_filter_expect=$2
 
```

Comment:
Can't we just use "filter" and "expect" ?

Reply:
local is supported in many shells, including bash, ksh, dash, and BusyBox ash. However, strictly speaking, it's not in POSIX. While trying to run we get the following warning :
```
In tests/shell/test_brstack.sh line 50:
	local filter=$1
        ^----------^ SC3043 (warning): In POSIX sh, 'local' is undefined.
```

Reply:
What athira trying to say is, why u changed variable name. Means, filter to test_filter_filter and same for expect

Reply:
local is supported in many shells, including bash, ksh, dash, and BusyBox ash. However, strictly speaking, it's not in POSIX. While trying to run we get the following warning :
```
In tests/shell/test_brstack.sh line 50:
	local filter=$1
        ^----------^ SC3043 (warning): In POSIX sh, 'local' is undefined.
```
In POSIX sh, you can adopt some convention to avoid accidentally overwriting variables names, e.g. prefixing with the function name like:
```
myfunc() {
  _myfunc_i=0
```
this is what I have adopted in the above piece of code to fix the POSIX warnings